In [1]:
import sys
sys.path.append(r"C:\Users\tlili\OneDrive\Bureau\Bootcamp\AI-powered-cryptocurrency-market-analysis-and-decision-support-system")

from news.llm import full_analysis

result = full_analysis(
    question = "What will Bitcoin do tomorrow based on current market conditions?",
    coin     = "btc"
)

print(result["analysis"])


  FULL ANALYSIS PIPELINE — BTC

⚙️  Step 1 : Running ML model...

  LIVE PIPELINE — BTC
  Warmup window : 2025-10-18 → 2026-02-15

📥 Fetching crypto prices...
  ✅ BTC-USD: 119 rows

📥 Fetching macro signals...
  ✅ SPY: 81 rows | 2025-10-20 → 2026-02-13
  ✅ DX-Y.NYB: 81 rows | 2025-10-20 → 2026-02-13
  ✅ ^VIX: 81 rows | 2025-10-20 → 2026-02-13

📥 Fetching Fear & Greed...
  ✅ Fear & Greed: 150 rows | 2025-09-19 → 2026-02-15

⚙️  Computing macro returns...

⚙️  Merging signals...
  Shape after merge: (119, 9)

⚙️  Preprocessing...

⚙️  Computing features...
  └─ Feature engineering complete ✅

  Date        : 2026-02-13
  Price       : $68,857.84
  Features    : 21 ✅
  NaN check   : 0 (should be 0)

  PREDICTION — BTC
  Date       : 2026-02-15
  Direction  : UP 📈
  Confidence : 70.54%
  Model      : AI Prediction Model

⚙️  Step 2 : Fetching news...

  TAVILY NEWS FETCHER — BTC
  Date : 2026-02-15

  🔍 Searching BTC news...
    'Bitcoin BTC price market analysis today' → 5 articles
    '

The full pipeline works end to end 
ML prediction → News fetching → RAG retrieval → LLM analysis
What the LLM produced:

- Correctly cited the 70.54% confidence
- Found the $69,536 price and ETF data from news
- Identified the ETF drain as a risk factor
- Added the financial disclaimer automatically
- Followed the exact structure we defined